Assets/Scripts/PlantSpawner.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections;
+using UnityEngine;
+
+namespace Assets.Scripts
+{
+    public class PlantSpawner : MonoBehaviour
+    {
+
+        public Terrain WorldTerrain;
+        public LayerMask TerrainLayer;
+        public static float TerrainLeft, TerrainRight, TerrainTop, TerrainBottom, TerrainWidth, TerrainLength, TerrainHeight;
+
+        public static ArrayList units = new ArrayList();
+        public static ArrayList positions = new ArrayList();
+        public static ArrayList rotations = new ArrayList();
+
+
+        private void Awake()
+        {
+            TerrainLeft = WorldTerrain.transform.position.x;
+            TerrainBottom = WorldTerrain.transform.position.z;
+            TerrainWidth = WorldTerrain.terrainData.size.x;
+            TerrainLength = WorldTerrain.terrainData.size.z;
+            TerrainHeight = WorldTerrain.terrainData.size.y;
```

Comment:
22, 23, 24 are all trying access non existant terrain data on world terrain and this is causing the plants not to spawn